packages/website/src/components/Playground.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useState } from 'react';
+import type Monaco from 'monaco-editor';
+import useThemeContext from '@theme/hooks/useThemeContext';
+
+import styles from './Playground.module.css';
+import Loader from './layout/Loader';
+
+import useHashState from './hooks/useHashState';
+import OptionsSelector from './OptionsSelector';
+import ASTViewer from './ast/ASTViewer';
+import clsx from 'clsx';
+import Editor from './editor/Editor';
+import { findNode } from './lib/selection';
+import type { RuleDetails } from './types';
+
+import type {
+  ParseForESLintResult,
+  TSESTree,
+} from '@typescript-eslint/website-eslint';
+
+function Playground(): JSX.Element {
```

Comment:
This is a lot of files and is pretty fully featured. I'll do my best to review (and recognize having the non-MVP features in is useful for seeing where things are going)... but it would be easier if features like copying markdown, showing ASTs, and modifying configs as more than just JSON were split into separate PRs.

Not a blocker, and still shy of a request 😄

Reply:
i'm actually not sure about config formats, currently i'm accepting reading `compilerOptions` and `rules` from config